### PR TITLE
feat(ckerc20): NNS proposal to add ckEURC

### PR DESCRIPTION
**Pull request by ninegua**
(XC-169): NNS proposal to add ckEURC, the twin token of [EURC](https://etherscan.io/token/0x1abaea1f7c830bd89acc67ec4af516284b1bc33c).

### CODE REVIEW

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
```
Comment (chmllr) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md:24]: - The logo checked.
- Contract address `0x1aBaEA1f7C830bD89Acc67eC4af516284b1bC33c` checked.

```
4449444c0e6b03ba9edbce040186dac7c60906c3cab9d20a0b6c05b59798d00202e898cad20405e9c2d6a70905a1eedde50a0598c6c2cf0b056e036c04e7aba21404cdd1f4e20404e4d8cce80b0494e2c9d00c046e7d6e716c03b59798d00207ffb08aab0809cb83d3c90f0a6e086c04e7aba2147dcdd1f4e2047de4d8cce80b7d94e2c9d00c7d6d686e686c0292bec7bd030cfdb5c9a9070d6c0299e7a4cf087db4e3ade809716c05c295a993017b9efeb9a40371f2c794ae037db1a38bf40d7191c9aafe0d71010002012a3078316142614541316637433833306244383941636336376543346166353136323834623162433333630606636b45555243904eca28646174613a696d6167652f7376672b786d6c3b6261736536342c50484e325a79423361575230614430694d7a59774969426f5a576c6e61485139496a4d324d434967646d6c6c64304a76654430694d43417749444d324d43417a4e6a416949475a7062477739496d3576626d5569494868746247357a50534a6f644852774f693876643364334c6e637a4c6d39795a7938794d4441774c334e325a79492b436a786e49474e7361584174634746306144306964584a734b434e6a62476c774d4638354e7a6c664e545570496a344b50484268644767675a443069545445344d434177517a49334f533430494441674d7a5977494467774c6a59674d7a5977494445344d454d7a4e6a41674d6a63354c6a51674d6a63354c6a51674d7a5977494445344d43417a4e6a42444f4441754e69417a4e6a41674d4341794e7a6b754e434177494445344d454d77494467774c6a59674f4441754e694177494445344d434177576949675a6d6c7362443069497a4e434d4442434f53497650676f38634746306143426d615778734c584a3162475539496d56325a5735765a47516949474e7361584174636e56735a5430695a585a6c626d396b5a4349675a443069545451774c6a51674d546b774c6a52444e4455754e4341794e546b754e4341784d4441754e69417a4d5451754e6941784e6a6b754e69417a4d546b754e6c597a4d7a55754d6b4d354d69417a4d7a41674d7a41674d6a5934494449304c6a67674d546b774c6a52494e4441754e466f6949475a7062477739496e56796243676a63474670626e517758327870626d5668636c38354e7a6c664e5455704969382b436a78775958526f49475a7062477774636e56735a5430695a585a6c626d396b5a43496759327870634331796457786c50534a6c646d56756232526b4969426b50534a4e4d5459354c6a59674e4441754e454d784d4441754e6941304e533430494451314c6a51674d5441774c6a59674e4441754e4341784e6a6b754e6b67794e433434517a49354c6a67674f5449674f5449674d6a6b754f4341784e6a6b754e6941794e433434566a51774c6a52614969426d615778735053496a4d6a6c42516b55794969382b436a78775958526f49475a7062477774636e56735a5430695a585a6c626d396b5a43496759327870634331796457786c50534a6c646d56756232526b4969426b50534a4e4d7a45354c6a59674d5459354c6a52444d7a45304c6a59674d5441774c6a51674d6a55354c6a51674e4455754d6941784f5441754e4341304d433479566a49304c6a5a444d6a5934494449354c6a67674d7a4d774c6a49674f5445754f43417a4d7a55754d6941784e6a6b754e45677a4d546b754e6c6f6949475a7062477739496e56796243676a63474670626e517858327870626d5668636c38354e7a6c664e5455704969382b436a78775958526f49475a7062477774636e56735a5430695a585a6c626d396b5a43496759327870634331796457786c50534a6c646d56756232526b4969426b50534a4e4d546b774c6a51674d7a45354c6a5a444d6a55354c6a51674d7a45304c6a59674d7a45304c6a59674d6a55354c6a51674d7a45354c6a59674d546b774c6a52494d7a4d314c6a4a444d7a4d774c6a49674d6a5934494449324f43417a4d7a41674d546b774c6a51674d7a4d314c6a4a574d7a45354c6a5a614969426d615778735053496a4d6a6c42516b55794969382b436a78775958526f49475a7062477774636e56735a5430695a585a6c626d396b5a43496759327870634331796457786c50534a6c646d56756232526b4969426b50534a4e4d6a457a4c6a45354e7941334e5334344d7a5179517a49774f533435494463304c6a63344d446b674d6a41334c6a49774e4341334e6934334d7a6334494449774e7934794d4451674f4441754d546b354e4659344f4334334d544930517a49774e7934794d4451674f5445754d444d7a4e4341794d4467754f54557a49446b7a4c6a59334e5341794d5445754d544d7949446b304c6a51334d7a5a444d6a51324c6a45794f4341784d4463754d6a6b79494449334d5334784e7a45674d5451774c6a6b7a494449334d5334784e7a45674d5467774c6a4d784d6b4d794e7a45754d546378494449784f5334324f5451674d6a51324c6a45794f4341794e544d754d7a4d674d6a45784c6a457a4d6941794e6a59754d545135517a49774f4334334e4449674d6a59334c6a41794e4341794d4463754d6a4130494449324f53347a4e6a55674d6a41334c6a49774e4341794e7a45754f544578566a49344d4334304d6a4e444d6a41334c6a49774e4341794f444d754f446731494449774f533435494449344e5334344e4449674d6a457a4c6a45354e7941794f4451754e7a6735517a49314e7934304d6a49674d6a63774c6a59314f4341794f446b754e445133494449794f5334794d6a55674d6a67354c6a51304e7941784f4441754d7a4579517a49344f5334304e4463674d544d784c6a4d354f4341794e5463754e444979494467354c6a6b324e4459674d6a457a4c6a45354e7941334e5334344d7a5179576949675a6d6c736244306964326870644755694c7a344b50484268644767675a6d6c73624331796457786c50534a6c646d56756232526b4969426a62476c774c584a3162475539496d56325a5735765a47516949475139496b30784e5449754d7a6330494467774c6a45354f5452444d5455794c6a4d334e4341334e6934334d7a6334494445304f5334324e7a67674e7a51754e7a67774f5341784e4459754d7a6778494463314c6a677a4e444a444d5441794c6a45314e4341344f5334354e6a5132494463774c6a45794f5449674d544d784c6a4d354f4341334d4334784d6a6b79494445344d43347a4d544a444e7a41754d5449354d6941794d6a6b754d6a4932494445774d6934784e5451674d6a63774c6a59314f4341784e4459754d7a6778494449344e4334334f446c444d5451354c6a59334f4341794f4455754f445179494445314d69347a4e7a51674d6a677a4c6a67344e5341784e5449754d7a6330494449344d4334304d6a4e574d6a63784c6a6b784d554d784e5449754d7a6330494449324f5334314f446b674d5455774c6a59794e4341794e6a59754f545133494445304f4334304e4455674d6a59324c6a45304f554d784d544d754e445135494449314d79347a4d7941344f4334304d445532494449784f5334324f544d674f4467754e4441314e6941784f4441754d7a4579517a67344c6a51774e5459674d5451774c6a6b7a494445784d7934304e446b674d5441334c6a49354d6941784e4467754e44513149446b304c6a51334d7a5a444d5455774c6a59794e4341354d7934324e7a55674d5455794c6a4d334e4341354d5334774d7a4d30494445314d69347a4e7a51674f4467754e7a45794d3159344d4334784f546b30576949675a6d6c736244306964326870644755694c7a344b50484268644767675a443069545449784e7934784f4341794d4467754f445132544449774f4334344d5455674d6a41314c6a45774f554d794d4459754f546330494449774e4334794f4459674d6a41304c6a67774d6941794d4455754d445531494449774d7934354d5341794d4459754f445931517a45354f5334354e6a4d674d6a45304c6a67334e7941784f544d754e444179494449784f5334354f444d674d5467304c6a637a4e5341794d546b754f54677a517a45334e6934344f4467674d6a45354c6a6b344d7941784e7a41754e546b31494449784e6934784e4341784e6a55754e7a6378494449774f43347a4e7a56444d54597a4c6a67344e6941794d4455754d7a63674d5459794c6a4d344e7941794d4449754d444533494445324d5334794e4463674d546b344c6a4d304d3067784f444d754e7a4133517a45344e5334784d7941784f5467754d7a517a494445344e6934304d7a49674d546b334c6a55304d5341784f4463754d446331494445354e6934794e3077784f5449754e444d7a494445344f5334774d7a64444d546b7a4c6a63774d5341784f4459754e544933494445354d5334344e7a63674d54677a4c6a55324d7941784f446b754d445933494445344d7934314e6a4e494d5455344c6a63784f554d784e5467754e6a4d78494445344d6934774d4463674d5455344c6a55334f4341784f4441754e4449674d5455344c6a55334f4341784e7a67754e7a6731517a45314f4334314e6a45674d5463334c6a45324e5341784e5467754e6a4133494445334e5334314e7a63674d5455344c6a59354d5341784e7a51754d44457a534445344d7934334d4464444d5467314c6a457a494445334e4334774d544d674d5467324c6a517a4d6941784e7a4d754d6a4579494445344e7934774e7a55674d5463784c6a6b304d5577784f5449754e444d7a494445324e4334334d4468444d546b7a4c6a63774d6941784e6a49754d546b34494445354d5334344e7a63674d5455354c6a497a4e4341784f446b754d445933494445314f5334794d7a52494d5459784c6a49784f554d784e6a55754d6a6b79494445304e5334354e5451674d54637a4c6a63354e5341784d7a63754d6a5530494445344e4334334d7a55674d544d334c6a51794e554d784f544d754d4463794944457a4e7934304d6a55674d546b354c6a55304d7941784e4445754f546735494449774d7934324f5455674d5451354c6a51324e454d794d4451754e6a4931494445314d5334784d7a6b674d6a41324c6a63784f5341784e5445754e7a6778494449774f4334304e7a45674d5455784c6a41774f4577794d5459754f446335494445304e7934794f545a444d6a45344c6a6735494445304e6934304d4467674d6a45354c6a63304e6941784e444d754f546332494449784f4334324f4445674d5451794c6a41314d6b4d794d5441754f5449674d5449344c6a417a4e6941784f546b754e6a4131494445794d4334354f5455674d5467304c6a637a4e5341784d6a41754f546b31517a45334d5334784e6a59674d5449774c6a6b354e5341784e6a41754d6a6b30494445794e69347a4f5341784e5449754d444d354944457a4e7934774f5468444d5451334c6a49784e4341784e444d754e444131494445304d7934344e5455674d5455774c6a67774e5341784e4445754f445978494445314f5334794d7a52494d5449354c6a59304f554d784d6a67754d6a4931494445314f5334794d7a51674d5449324c6a6b794d7941784e6a41754d444d31494445794e6934794f4445674d5459784c6a4d774e6b77784d6a41754f544978494445324f4334314d7a6c444d5445354c6a59314e4341784e7a45754d445135494445794d5334304e7a67674d5463304c6a41784d7941784d6a51754d6a6735494445334e4334774d544e494d544d354c6a67304d304d784d7a6b754e7a5934494445334e5334314f444d674d544d354c6a59354e7941784e7a63754d5455334944457a4f5334324f5463674d5463344c6a63344e554d784d7a6b754e6a6333494445344d43347a4f546b674d544d354c6a63784e5341784f4445754f5467344944457a4f5334334f444d674d54677a4c6a55324d3067784d6a6b754e6a5135517a45794f4334794d6a55674d54677a4c6a55324d7941784d6a59754f54497a494445344e43347a4e6a55674d5449324c6a49344d5341784f4455754e6a4d31544445794d4334354d6a45674d546b794c6a67324f454d784d546b754e6a5530494445354e53347a4e7a67674d5449784c6a51334f4341784f5467754d7a517a494445794e4334794f446b674d546b344c6a4d304d3067784e4445754f444932517a45304e79347a4e7a49674d6a49784c6a49324e6941784e6a4d754d7a4d354944497a4e6934344f5467674d5467304c6a637a4e5341794d7a59754e546331517a45354f5334354d6a59674d6a4d324c6a55334e5341794d5445754d7a6332494449794f5334774d6a4d674d6a45354c6a4179494449784d7934354f445a444d6a45354c6a6b354d7941794d5449754d446378494449784f5334784e4341794d446b754e7a4979494449784e7934784f4341794d4467754f445132576949675a6d6c736244306964326870644755694c7a344b5043396e50676f385a47566d637a344b50477870626d5668636b6479595752705a57353049476c6b50534a7759576c756444426662476c755a574679587a6b334f5638314e53496765444539496a457a4d4334334d69496765544539496a4d774e4334784d69496765444939496a4d7a4c6a5134496942354d6a30694d6a49794c6a49794969426e636d466b61575675644656756158527a50534a31633256795533426859325650626c567a5a53492b436a787a644739774947396d5a6e4e6c644430694d4334794d534967633352766343316a62327876636a3069493056454d5555334f53497650676f3863335276634342765a6d5a7a5a585139496a456949484e3062334174593239736233493949694d314d6a49334f4455694c7a344b504339736157356c59584a48636d466b615756756444344b50477870626d5668636b6479595752705a57353049476c6b50534a7759576c756444466662476c755a574679587a6b334f5638314e53496765444539496a4d774f53347a4d69496765544539496a45794d7934774e69496765444939496a49784d6934774f43496765544939496a51784c6a45324969426e636d466b61575675644656756158527a50534a31633256795533426859325650626c567a5a53492b436a787a644739774947396d5a6e4e6c644430694d4334794d534967633352766343316a62327876636a3069493059784e5545794e43497650676f3863335276634342765a6d5a7a5a585139496a41754e6a676949484e3062334174593239736233493949694e47516b49774d3049694c7a344b504339736157356c59584a48636d466b615756756444344b50474e73615842515958526f49476c6b50534a6a62476c774d4638354e7a6c664e54556950676f38636d566a6443423361575230614430694d7a59774969426f5a576c6e61485139496a4d324d4349675a6d6c736244306964326870644755694c7a344b5043396a62476c77554746306144344b5043396b5a575a7a50676f384c334e325a7a344b06636b45555243
```

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Wasm Verification
+
+Verify that the hash of the gzipped WASM matches the proposed hash.
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+./gitlab-ci/container/build-ic.sh -c
+sha256sum ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
```
Comment (chmllr) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md:41]: Can't verify the build with these commands, as the build fails with

```
FAILED: Build did NOT complete successfully (594 packages loaded, 14939 targets configured)
    currently loading: @crate_index__lzma-sys-0.1.20//
    Fetching repository @crate_index__icu_normalizer_data-1.5.0; starting
    Fetching repository @crate_index__icu_provider_macros-1.5.0; Restarting.
    Fetching repository @crate_index__writeable-0.5.5; Restarting.
    Fetching repository @crate_index__icu_locid-1.5.0; Restarting.
    Fetching https://static.crates.io/crates/zerofrom/0.1.4/download
    Fetching https://static.crates.io/crates/yoke/0.7.4/download
    Fetching https://static.crates.io/crates/zerovec-derive/0.10.3/download
ERROR: /ic/rs/sns/swap/proto_library/BUILD.bazel:24:13: in srcs attribute of rust_library rule //rs/sns/swap/proto_library:proto_library: source file '//rs/sns/swap/proto_library:src/gen' is misplaced here (expected .rs)
ERROR: /ic/rs/sns/swap/proto_library/BUILD.bazel:24:13: in srcs attribute of rust_library rule //rs/sns/swap/proto_library:proto_library: '//rs/sns/swap/proto_library:src/gen' does not produce any rust_library srcs files (expected .rs)
ERROR: /ic/rs/sns/swap/proto_library/BUILD.bazel:24:13: Analysis of target '//rs/sns/swap/proto_library:proto_library' failed
ERROR: Analysis of target '//publish/canisters:canisters' failed; build aborted:
```

Any idea what to do here?

**Review by chmllr — APPROVED** [review]
Approving modulo the build hash reproduction (which had to be verified in the last release).

**Comment by THLO** [comment]
> (XC-169): NNS proposal to add ckERUC, the twin token of [EURC](https://etherscan.io/token/0x1abaea1f7c830bd89acc67ec4af516284b1bc33c).

It should be ckEURC.   :slightly_smiling_face:

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md (hunk 1 above)
Comment (THLO) [reply rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md:24]: The logo and the contract address look good to me too.

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md (hunk 2 above)
Comment (THLO) [reply rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_15.md:41]:
```
$ sha256sum ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
ccda8966249bc7d89f14968a242d6fb2c33d2363336303ec4f12977cab74c6f7  ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
```
Looks good!

**Review by THLO — APPROVED** [review]

**Pull request merged**